src/containerbuddy/main.go

```diff
@@ -48,10 +48,21 @@ func main() {
 		// Run our main application and capture its stdout/stderr.
 		// This will block until the main application exits and then os.Exit
 		// with the exit code of that application.
-		code, err := run(flag.Args())
+		config.Command = getCmd(flag.Args())
+		code, err := executeAndWait(config.Command)
 		if err != nil {
 			log.Println(err)
 		}
+		deregisterServices(config)
```

Comment:
Right, but as-written you're running the application and then running deregister afterwards. Which means the application exits and can't serve requests but the discovery service will still think it's healthy.